CHANGELOG.md

```diff
@@ -13,6 +13,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - Added `oe_sgx_get_signer_id_from_public_key()` function which helps a verifier of SGX
   reports extract the expected MRSIGNER value from the signer's public key PEM certificate.
+- OE SDK can now be built for Simulation Mode on non SGX x64 systems by passing HAS_QUOTE_PROVIDER=off.
```

Comment:
As worded, "built for Simulation Mode" sounds like you cannot use simulation mode at all without building this way.  Did you mean "built to run only on Simulation Mode"?

Also, I'm curious why you have to build it differently in the first place. If you have to have an SGX capable machine to _build_ on, that should be explained somewhere, since that is very surprising. Is that answered in some other doc?

Reply:
How about:
" OE SDK can now be built on a non SGX x64 Windows machine by passing HAS_QUOTE_PROVIDER=off. Previously this was not possible due to the hard dependency on sgx_enclave_common.dll which is available only on SGX machines."

The biggest impact this PR has is on the Windows experience. Previously building the SDK had a hard-dependency on sgx_enclave_common.dll which is installed via Windows Update only on SGX machines.

> If you have to have an SGX capable machine to build on, that should be explained somewhere, since that is very surprising

All our GettingStarted docs for Contributors list an SGX1 capable machine as a prerequisite.

Reply:
I have reworded the changelog entry.

Reply:
> All our GettingStarted docs for Contributors list an SGX1 capable machine as a prerequisite.

Good point.  Shouldn't this PR then update those docs?

Reply:
I just found PR #3158

Reply:
This is the most recent changelog description
`OE SDK can now be built and run in simulation mode on a non SGX x64 Windows machine by passing HAS_QUOTE_PROVIDER=off.
  Previously, the build would work, but running applications would fail due to missing sgx_enclave_common.dll.`

I have added a doc (#3155) for the  workflow that this PR enables - build the SDK for on non SGX x64 Windows machines and run Simulation Mode. Previously that was not possible.

As to 
>> All our GettingStarted docs for Contributors list an SGX1 capable machine as a prerequisite.

> Good point. Shouldn't this PR then update those docs?

We are not yet there. But making good progress towards it. An official build of the SDK does not use HAS_QUOTE_PROVIDER=off  (ie it requires HAS_QUOTE_PROVIDER=on) and does indeed require an SGX machine currently.

Reply:
I believe this PR also addressed the technical aspect of #2717 #2728 #1983. But I need to carefully read those issues and then write the missing documentation. Hence I'm trying to get the source code changes in first.